pkg/controller/daemon/daemoncontroller.go

```diff
@@ -579,23 +735,37 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) error {
 			}
 		}
 	}
-	errors := dsc.syncNodes(ds, podsToDelete, nodesNeedingDaemonPods)
+
+	cur, old, err := dsc.constructHistory(ds)
+	if err != nil {
+		return "", fmt.Errorf("failed to construct revisions of DaemonSet: %v", err)
+	}
+
+	err = dsc.cleanupHistory(ds, old, nodeToDaemonPods)
```

Comment:
I only moved history cleanup for Deployments so we could unblock Deployments broken by hash collisions. I think it's strictly more correct to cleanup at steady state, i.e., not before or during upgrades and I am planning on moving it back to its original place now that we have the hashing collision avoidance mechanism in place.